### PR TITLE
RATIS-1254. Add transfer leadership request and proto

### PR DESCRIPTION
**Pull request by runzhiwang**
## What changes were proposed in this pull request?

subtask of Support rolling upgrade and rollback: Add transfer leadership request and proto

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/RATIS-1254

## How was this patch tested?

No need ut.

### CODE REVIEW

**Comment by szetszwo** [comment]
Thanks @runzhiwang .  Will review this tomorrow.

File: ratis-proto/src/main/proto/Raft.proto [hunk 1]
```diff
@@ -369,6 +369,12 @@ message SetConfigurationRequestProto {
   repeated RaftPeerProto peers = 2;
 }
 
+// transfer leadership request
+message TransferLeadershipRequestProto {
+  RaftRpcRequestProto rpcRequest = 1;
+  RaftPeerProto targetPeer = 2;
```
Comment (szetszwo) [inline ratis-proto/src/main/proto/Raft.proto:375]: Let's name it newLeader.

File: ratis-client/src/main/java/org/apache/ratis/client/RaftClient.java [hunk 2]
```diff
@@ -68,6 +68,9 @@
   /** Send set configuration request to the raft service. */
   RaftClientReply setConfiguration(RaftPeer[] serversInNewConf) throws IOException;
 
+  /** Transfer leadership to the given server.*/
+  RaftClientReply transferLeadership(RaftGroupId group, RaftPeerId target) throws IOException;
```
Comment (szetszwo) [inline ratis-client/src/main/java/org/apache/ratis/client/RaftClient.java:72]: Let's rename the parameter to "newLeader".

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.  Just some minor comments on the parameter names.

**Comment by runzhiwang** [comment]
@szetszwo Thanks for review. CI passed, and I merge it.

**Comment by amaliujia** [comment]
+1

**Pull request merged**